kanidm_book/src/examples/k8s_ingress_example.md

```diff
@@ -6,69 +6,132 @@ Guard your Kubernetes ingress with Kanidm authentication and authorization.
 
-- [Kanidm](installing_the_server.html)
-- [Kubernetes v1.23](https://docs.k0sproject.io/v1.23.6+k0s.2/install/)
+- [Kanidm](../installing_the_server.html)
+- [Kubernetes v1.23 or above](https://docs.k0sproject.io/v1.23.6+k0s.2/install/)
 - [Nginx Ingress](https://kubernetes.github.io/ingress-nginx/deploy/)
+- [A Domain Name](https://domains.google/)
 - [CertManager - With Cluster Issuer](https://cert-manager.io/docs/installation/)
 - [ShellCheck](https://github.com/koalaman/shellcheck#user-content-installing)
 
 You can set up your certificates manually instead of using CertManager if you are comfortable doing so. If it is your first time setting up CertManager, take your time with installation and validation. For the `http01` solver the ingress port (8089) needs to be accessible from the internet - DNS validation is recommended.
 
 ## Instructions
 
-1. Install the Kubernetes dashboard
-
-```shell
-kubectl create -f https://raw.githubusercontent.com/kubernetes/kops/master/addons/kubernetes-dashboard/v1.10.1.yaml
-```
-
-2. Create a script from the content below, replacing every `<string>` (drop the `<>`) with appropriate values, then run it. This will example users, groups and the OAUTH2 configuration.
+1. Create a script `deplopy.sh` from the content below, replacing every `<string>` (drop the `<>`) with appropriate values, then run it. This will create the user, group and OAUTH2 resource.
 
 ```shell
 #!/bin/sh
 
 # Analysis self for common errors.
 shellcheck ./*.sh || exit 1
 
+# Login
+echo "User: idm_admin"
+kanidm login --name idm_admin
+
 # Create User
-kanidm account create k8s_example_user
+kanidm account create k8s_example_user "K8s Example User" --name idm_admin
+kanidm account credential update k8s_example_user --name idm_admin
+> password
+> commit
+kanidm account person extend k8s_example_user --legalname "John K8s Doe" --mail "jkd@email.address" --name idm_admin
 
 # Create Group
-kanidm group create k8s_example_group
+kanidm group create k8s_example_group --name idm_admin
 
 # Attach User to Group
-kanidm group add_member k8s_example_group k8s_example_user
+kanidm group add_members k8s_example_group k8s_example_user --name idm_admin
 
 # Setup Oauth Resource
 displayname="The Example"
 name=<oauth2_rs_name>
 domain=<host>
+echo "User: admin"
 kanidm login --name admin
-kanidm system oauth2 create "$name" "$displayname" "$domain" --name admin
+kanidm system oauth2 create "$name" "$displayname" "https://$domain" --name admin
 kanidm system oauth2 create_scope_map "$name" k8s_example_group openid email profile --name admin
 kanidm system oauth2 get "$name" --name admin
 
 ## For <cookie_secret>
 echo "Generating cookie secret:"
-docker run -ti --rm python:3-alpine python -c 'import secrets,base64; print(base64.b64encode(base64.b64encode(secrets.token_bytes(16))));' 
+docker run -ti --rm python:3-alpine python -c 'import secrets,base64; print(base64.b64encode(base64.b64encode(secrets.token_bytes(16))));'
```

Comment:
This will print it without the `b''` bits:
```suggestion
docker run -ti --rm python:3-alpine python -c 'import secrets,base64; print(base64.b64encode(base64.b64encode(secrets.token_bytes(16))).decode("utf-8"));'
```